pymatgen/core/sites.py

```diff
@@ -467,11 +467,14 @@ def __eq__(self, other: object) -> bool:
         if not isinstance(other, Site):
             return NotImplemented
 
+        self_props = {k: v for k, v in self.properties.items() if k != "label"}
+        other_props = {k: v for k, v in other.properties.items() if k != "label"}
+
```

Comment:
I don't think we can ignore the label when comparing sites. Would be quite unexpected behavior for most users:

```
from pymatgen.core.sites import Site

s1 = Site("Fe", [0.25, 0.35, 0.45], properties={"label": 'foo'})
s2 = Site("Fe", [0.25, 0.35, 0.45], properties={"label": 'bar'})
s1 == s2
>>> True
```

Reply:
I am unsure what the site labels will be used for. But I would refrain from adding it to properties. When we talk about properties, we usually mean real physical quantities. E.g., magmom. Then two sites with different magmom must be unequal. I would instead use an attribute. E.g., just site.label = "Fe1". People who care will know to use it. Most people don't really care?

Reply:
Maybe creating an issue for a larger discussion would be useful, since I think this ultimately relates to `site_properties` not being typed, which caused a lot of downstream problems.

For example, imagine we had explicit types for possible site properties: `ScalarProperty`, `PolarVectorProperty`, `AxialVectorProperty`, etc, `Categorical`, `Comment`. Something physical like `Magmom` could be handled under this system, and also something like a label could be a `Comment` type, which could then be gracefully excluded from equality comparisons. Just a thought.

Reply:
I like @shyuep's suggestion of storing label directly as a site attribute instead of in properties. Perhaps the most sensible short-term solution.

We can open an issue to track building a `properties` ontology as suggested by @mkhorton later so we don't forget.

Reply:
This is the reason why I switched Species.properties to simply Species.spin in the recent commits (with some backward compatibility code). In the old days, I was thinking that we may want to put other properties on Species. It turns out no one wanted to do that. In the end, that may be how we go. Instead of Site.properties, we simply have proper attributes for everything. So Site.magmom, Site.whatever, Site.label. We then decide which of these attributes are checked for equality.

Reply:
Thanks for the suggestions. I have implemented this now as `site.label` as suggested. This change turned out to be a bit more substantial than I was hoping for.

Let me know if this approach works for you. If so, I'm willing to spend some time to complete the implementation for all methods on Structure that work with the sites list.

Reply:
Did you forget to push? Nothing for us to review at this point.

Reply:
I suggest you push your changes to this PR. We will look at it and give suggestions (or just simply merge). In the end, a label attribute is simply something that can be used by people who really want the site label. As I said, most users don't really care about the site labels in a CIF file. So this will be invisible to most people.